assets/js/components/HostDetails/HostDetails.jsx

```diff
@@ -36,6 +38,15 @@ function HostDetails() {
   const { grafanaPublicUrl } = config;
 
   const [exportersStatus, setExportersStatus] = useState([]);
+  const [cleanUpModalOpen, setCleanUpModalOpen] = useState(false);
+
+  const dispatch = useDispatch();
+  const navigate = useNavigate();
+
+  const cleanUpHost = (selectedHost) => {
+    setCleanUpModalOpen(false);
+    dispatch(deregisterHost({ host: selectedHost, navigate }));
```

Comment:
We careful with this!
If you change the payload format for the function, you need to change it every place that it is used, the host list view in this case.
I would suggest to go with `deregisterHost({ ...selectedHost, navigate })`, but if you prefer this way don't forget to change the host list view call as well

Reply:
Most probably not...